### ISSUE TITLE
Homebrew build fails on OSX 10.9.5

### ISSUE THREAD

**Issue by robertgzr** [saldl/saldl#5]
```
$ brew install https://github.com/saldl/saldl/raw/master/OSX/saldl.rb
######################################################################## 100.0%
==> Downloading https://github.com/saldl/saldl/archive/v31.tar.gz
==> Downloading from https://codeload.github.com/saldl/saldl/tar.gz/v31
######################################################################## 100.0%
==> ./waf configure --libcurl-cflags=-I/usr/local/opt/curl/include --libcurl-libs=-L/usr/local/opt/curl/lib -lcurl --prefix=/usr/local/Cellar/saldl/31 --saldl-version=v31
==> ./waf build
Last 15 lines from /Users/user/Library/Logs/Homebrew/saldl/02.waf:
[12/17] Compiling src/utime.c
[13/17] Compiling src/transfer.c
[14/17] Compiling src/saldl.c
[15/17] Compiling src/main.c
[16/17] Compiling saldl.1.txt
  File "/usr/local/opt/asciidoc/bin/a2x", line 76
    print '%s: %s' % (PROG,msg)
                 ^
SyntaxError: invalid syntax

Waf: Leaving directory `/private/tmp/saldl20151108-40246-1xpp3fx/saldl-31/build'
Build failed
 -> task in '' failed (exit status 1):
    {task 4325720920: a2x-man-1 saldl.1.txt -> saldl.1}
['/usr/local/opt/asciidoc/bin/a2x', '-f', 'manpage', '-D.', '../saldl.1.txt']
```

**Comment by MoSal** [comment]
I will have a look later today.

The error is in `a2x` (a part of asciidoc) which we use to build the manual. I think  python 3 is used somehow to run `a2x`. And the error shows that `a2x` uses python 2 syntax.

**Comment by MoSal** [comment]
@robertgzr 
What is the output of:

```
which python
`which python` --version
```

**Comment by robertgzr** [comment]
`Python 3.5.0`

Which version does it expect?
edit: nevermind... It wants 2.x right? Just read your previous comment...

**Comment by MoSal** [comment]
@robertgzr 
`python` should always point to 2.x [according to the docs](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Homebrew-and-Python.md#python-2x-or-python-3x).

What is the output of `which python`?
Was that file installed by Homebrew?

**Comment by robertgzr** [comment]
@MoSal 
under `/usr/local/bin/python`
I switched it back to `2.7.10` now and I still get that same error

Oh and yes that was installed by homebrew and it's definitely in font of `/usr/bin/python` (v3.4.3)

**Comment by MoSal** [comment]
@robertgzr 
Now I'm confused.
I just installed `python3` from Homebrew. And it definitely does not install a `python` executable.
And OSX never shipped Python 3 as far as I know. Where did your `python 3.4.3` come from?

Anyway, you should definitely **not** get that error with `python 2.7`. You can temporarily change the first line of `/usr/local/opt/asciidoc/bin/a2x` from:

```
#!/usr/bin/env python
```

To:

```
#!/usr/bin/env python2.7
```

and see what happens.

**Comment by robertgzr** [comment]
@MoSal 
For now I got it installed by including the `--disable-man` flag in `./waf configure`
I tried editing `a2x` but that didn't work for me

I will look at the whole python/asciidoc thing again later and try to fix this... I'll let you know if I find something

**Comment by MoSal** [comment]
Good.

Let me know If I need to change something in the formula.

**Comment by robertgzr** [comment]
@MoSal 
I don't know if it is possible or worth it to include the option to build without the manpage as it is trivial to edit the formula to do that by hand.

Btw I tried compiling from the github source directly and manually setting the configure flags brew uses but it complained about `-lcurl`

**Comment by MoSal** [comment]
---

> I don't know if it is possible or worth it to include the option to build without the manpage as it is trivial to edit the formula to do that by hand.

Yeah. I don't think adding an option is necessary.
Building the manual should just work if python versions are installed correctly.

---

> Btw I tried compiling from the github source directly and manually setting the configure flags brew uses but it complained about `-lcurl`

```
brew info curl
```

Did you pass `--libcurl-cflags="-I/usr/local/opt/curl/include" --libcurl-libs="-L/usr/local/opt/curl/lib -lcurl"` to `./waf configure`?

What is the exact error message?

**Comment by robertgzr** [comment]
@MoSal 
**So I was able to build it successfully without brew.**

I reinstalled both pythons (python (2.7.10 and 3.5.0 via brew)
That didn't work...
Going back to the config flags after I saw your message I realized that I didn't put the `""` around the `--libcurl-libs` so thats why it was complaining about `-lcurl`:

`./waf configure --libcurl-cflags="-I/usr/local/opt/curl/include" --libcurl-libs="-L/usr/local/opt/curl/lib -lcurl"`

I just don't understand why the error was related to python and the manpages?

Now I'm trying to get that brew formula to work for me:
I put the `""` into formula and it fails with something else:

```
==> ./waf configure --libcurl-cflags="-I/usr/local/opt/curl/include" --libcurl-libs="-L/usr/local/opt/curl/lib -lcurl" --prefix=/usr/local/Cellar/saldl/31 --saldl-version=v31
==> ./waf build
Last 15 lines from /Users/robertgz/Library/Logs/Homebrew/saldl/02.waf:

Waf: Leaving directory `/private/tmp/saldl20151109-98864-1ilysq9/saldl-31/build'
Build failed
 -> task in 'saldl-objs' failed (exit status 1):
    {task 4334749736: c progress.c -> progress.c.1.o}
['clang', '-Wall', '-Wextra', '-Wmissing-format-attribute', '-Wno-newline-eof', '-Werror', '-std=gnu99', '-pedantic', '-fPIE', '"-I/usr/local/opt/curl/include"', '-I/usr/local/Cellar/libevent/2.0.22/include', '-DSALDL_VERSION="v31"', '-D_FILE_OFFSET_BITS=64', '-D_GNU_SOURCE', '-DHAVE_MKDIR=1', '-DHAVE_GETTIMEOFDAY=1', '-DHAVE_FILENO=1', '-DHAVE_ISATTY=1', '-DHAVE_IOCTL=1', '-DHAVE_ASPRINTF=1', '-DHAVE_STRCASESTR=1', '-DHAVE_STRSIGNAL=1', '-DHAVE_SIGACTION=1', '-DHAVE_SIGADDSET=1', '-DHAVE_LIBEVENT_PTHREADS=1', '-D_THREAD_SAFE', '../src/progress.c', '-c', '-o', '/private/tmp/saldl20151109-98864-1ilysq9/saldl-31/build/src/progress.c.1.o']
 -> task in 'saldl-objs' failed (exit status 1):
    {task 4334749848: c events.c -> events.c.1.o}
['clang', '-Wall', '-Wextra', '-Wmissing-format-attribute', '-Wno-newline-eof', '-Werror', '-std=gnu99', '-pedantic', '-fPIE', '"-I/usr/local/opt/curl/include"', '-I/usr/local/Cellar/libevent/2.0.22/include', '-DSALDL_VERSION="v31"', '-D_FILE_OFFSET_BITS=64', '-D_GNU_SOURCE', '-DHAVE_MKDIR=1', '-DHAVE_GETTIMEOFDAY=1', '-DHAVE_FILENO=1', '-DHAVE_ISATTY=1', '-DHAVE_IOCTL=1', '-DHAVE_ASPRINTF=1', '-DHAVE_STRCASESTR=1', '-DHAVE_STRSIGNAL=1', '-DHAVE_SIGACTION=1', '-DHAVE_SIGADDSET=1', '-DHAVE_LIBEVENT_PTHREADS=1', '-D_THREAD_SAFE', '../src/events.c', '-c', '-o', '/private/tmp/saldl20151109-98864-1ilysq9/saldl-31/build/src/events.c.1.o']
 -> task in 'saldl-objs' failed (exit status 1):
    {task 4334749960: c write_modes.c -> write_modes.c.1.o}
['clang', '-Wall', '-Wextra', '-Wmissing-format-attribute', '-Wno-newline-eof', '-Werror', '-std=gnu99', '-pedantic', '-fPIE', '"-I/usr/local/opt/curl/include"', '-I/usr/local/Cellar/libevent/2.0.22/include', '-DSALDL_VERSION="v31"', '-D_FILE_OFFSET_BITS=64', '-D_GNU_SOURCE', '-DHAVE_MKDIR=1', '-DHAVE_GETTIMEOFDAY=1', '-DHAVE_FILENO=1', '-DHAVE_ISATTY=1', '-DHAVE_IOCTL=1', '-DHAVE_ASPRINTF=1', '-DHAVE_STRCASESTR=1', '-DHAVE_STRSIGNAL=1', '-DHAVE_SIGACTION=1', '-DHAVE_SIGADDSET=1', '-DHAVE_LIBEVENT_PTHREADS=1', '-D_THREAD_SAFE', '../src/write_modes.c', '-c', '-o', '/private/tmp/saldl20151109-98864-1ilysq9/saldl-31/build/src/write_modes.c.1.o']
 -> task in 'saldl-objs' failed (exit status 1):
    {task 4334750072: c queue.c -> queue.c.1.o}
['clang', '-Wall', '-Wextra', '-Wmissing-format-attribute', '-Wno-newline-eof', '-Werror', '-std=gnu99', '-pedantic', '-fPIE', '"-I/usr/local/opt/curl/include"', '-I/usr/local/Cellar/libevent/2.0.22/include', '-DSALDL_VERSION="v31"', '-D_FILE_OFFSET_BITS=64', '-D_GNU_SOURCE', '-DHAVE_MKDIR=1', '-DHAVE_GETTIMEOFDAY=1', '-DHAVE_FILENO=1', '-DHAVE_ISATTY=1', '-DHAVE_IOCTL=1', '-DHAVE_ASPRINTF=1', '-DHAVE_STRCASESTR=1', '-DHAVE_STRSIGNAL=1', '-DHAVE_SIGACTION=1', '-DHAVE_SIGADDSET=1', '-DHAVE_LIBEVENT_PTHREADS=1', '-D_THREAD_SAFE', '../src/queue.c', '-c', '-o', '/private/tmp/saldl20151109-98864-1ilysq9/saldl-31/build/src/queue.c.1.o']
```

**Comment by MoSal** [comment]
The quotes you added are the problem. There was nothing wrong with formula as far as quoting goes.

I might have figured out what's going on here. Can you add these lines to the formula right before `system` calls:

``` ruby
puts `which python`
puts ENV["PATH"]
```

What is the output of brew before `==> ./waf configure` ?

**Comment by robertgzr** [comment]
Oh.

```
/usr/bin/python
/usr/local/Library/ENV/4.3:/usr/local/opt/pkg-config/bin:/usr/local/opt/asciidoc/bin:/usr/local/opt/curl/bin:/usr/local/opt/openssl/bin:/usr/local/opt/libevent/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

**Comment by MoSal** [comment]
Okay.

tl;dr
Add this line to the formula and the build should succeed:

``` ruby
ENV["PATH"] = "/usr/local/bin:" + ENV["PATH"]
```

Detailed explanation below.

---

So outside brew, `PATH` starts with `/usr/local/bin`. And that's the first path searched.
But homebrew seems to set `PATH` while running formulas in a way that excludes executables from unlisted dependencies. That's smart.

`/usr/bin/python` ,shipped by OSX, is supposed to be python 2. But it's not in your system for some reason. python versions installed via brew are not available while running the formula, as we are not listing them as dependencies. So, `/usr/local/bin` needs to be prepended  to `PATH` to make executables from those versions searched first.

**Comment by MoSal** [comment]
Adding python as a dependency would work too, I guess:

``` ruby
depends_on "python" => :build
```

**Comment by robertgzr** [comment]
Thanks!
Good to know that brew expects 2.x
I had somehow linked: 
`/System/Library/Frameworks/Python.framework/Versions/3.4/bin/python -> /usr/bin/python` 

When this is done, do you want me to submit the formula to homebrew?

**Comment by MoSal** [comment]
I think the formula is fine as is.
Feel free to submit.

I think the bug can be closed now.
